### PR TITLE
style: Fix dict-index-missing-items (PLC0206)

### PR DESCRIPTION
**Pull request by echoix**
Ruff rule: https://docs.astral.sh/ruff/rules/dict-index-missing-items/

9 instances solved

### CODE REVIEW

File: gui/wxpython/gmodeler/model.py [hunk 1]
```diff
@@ -2256,7 +2256,7 @@ def _processConditions(self):
             pos, size = self._getDim(node)
             text = self._filterValue(self._getNodeText(node, "condition")).strip()
             aid = {"if": [], "else": []}
-            for b in aid.keys():
+            for b in aid.keys():  # noqa: PLC0206
```
Comment (echoix) [inline gui/wxpython/gmodeler/model.py:2259]: I was conservative here. I'm not sure using `key, value` would behave when appending to `value` during iteration.

File: gui/wxpython/datacatalog/tree.py [hunk 2]
```diff
@@ -1677,9 +1677,9 @@ def DisplayLayer(self):
             all_names.append(name)
         # if self.selected_location[0].data['name'] == gisenv()['LOCATION_NAME'] and
         # self.selected_mapset[0]:
-        for ltype in names:
-            if names[ltype]:
-                self._giface.lmgr.AddMaps(list(reversed(names[ltype])), ltype, True)
+        for ltype, value in names.items():
+            if value:
+                self._giface.lmgr.AddMaps(list(reversed(value)), ltype, True)
```
Comment (echoix) [inline gui/wxpython/datacatalog/tree.py:1682]: I'm open to a better name than `value`, I didn't manage to infer from context.

File: python/grass/temporal/temporal_granularity.py [hunk 3]
```diff
@@ -466,7 +466,7 @@ def compute_absolute_time_granularity(maps):
         # start time is required in TGIS and expected to be present
         if end:
             map_datetime_delta = compute_datetime_delta(start, end)
-            for time_unit in granularity_units:
+            for time_unit in granularity_units.keys():  # noqa: PLC0206
```
Comment (echoix) [inline python/grass/temporal/temporal_granularity.py:469]: Even once fixed, the rule still triggered.

File: python/grass/temporal/temporal_granularity.py [hunk 4]
```diff
@@ -484,7 +484,7 @@ def compute_absolute_time_granularity(maps):
             else:
                 gap_datetime_delta = compute_datetime_delta(previous_start, start)
             # Add to the set of the smallest granularity in the granularity_units dict
-            for time_unit in granularity_units:
+            for time_unit in granularity_units.keys():  # noqa: PLC0206
```
Comment (echoix) [inline python/grass/temporal/temporal_granularity.py:487]: Here also, even once fixed, the rule still triggered.

File: scripts/i.pansharpen/i.pansharpen.py [hunk 5]
```diff
@@ -732,7 +732,7 @@ def matchhist(original, target, matched):
         stats = gs.decode(stats_out.communicate()[0]).split("\n")[:-1]
         stats_dict = dict(s.split(":", 1) for s in stats)
         total_cells = 0  # total non-null cells
-        for j in stats_dict:
+        for j in stats_dict.keys():  # noqa: PLC0206
```
Comment (echoix) [inline scripts/i.pansharpen/i.pansharpen.py:735]: Rule still triggered after fixing.

File: scripts/r.pack/r.pack.py [hunk 6]
```diff
@@ -130,11 +130,11 @@ def main():
 
         # Copy vrt files
         if vrt_files:
-            for f in vrt_files.keys():
+            for f, value in vrt_files.keys():
```
Comment (echoix) [inline scripts/r.pack/r.pack.py:133]: I'm open to a better name than `value`, I didn't manage to infer from context.

**Review by ninsbl — CHANGES_REQUESTED** [review]
I think there is one glitch.

In general I see you explicitly call `.keys()` when you loop over dicts. Should that be the standard?

Weird that ruff complains despite your fixes. Is there probably an issue with ruff?

**Comment by echoix** [comment]
> I think there is one glitch.
> 
> 
> 
> In general I see you explicitly call `.keys()` when you loop over dicts. Should that be the standard?

There are three ways (and one more): `.keys()` for only returning the dict's keys, `.values()` for only returning the dict's values, and `.items()` to return a tuple of both. Using the appropriate one when not using both of them is the correct thing to do.

(The last one is without anything, that returns the keys if I remember right, but it's not as obvious for reading. It's more useful for `in` checks.)



> 
> Weird that ruff complains despite your fixes. Is there probably an issue with ruff?
Probably I should be reporting it and getting it fixed in a week or so. That one is a preview rule I think.

**Review by ninsbl — APPROVED** [review]

**Comment by echoix** [comment]
@ninsbl I assume you were fine with the comments I added as explanations, so I’ll resolve them

**Pull request merged**